From 0af36cc0dcf5a6654e25eb33509fe0bc44a1dd81 Mon Sep 17 00:00:00 2001
From: Rajendra Nayak <rnayak@ti.com>
Date: Tue, 2 Mar 2010 17:25:30 +0530
Subject: [PATCH 6/6] OMAP: Serial: Check wk_st only if present

Uart on the resume path tries to read wk_st registers, even
on architectures were its not present/populated.
This patch fixes the issue.

Signed-off-by: Rajendra Nayak <rnayak@ti.com>
---
 arch/arm/mach-omap2/serial.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/serial.c b/arch/arm/mach-omap2/serial.c
index 1ac361b..a0046ce 100644
--- a/arch/arm/mach-omap2/serial.c
+++ b/arch/arm/mach-omap2/serial.c
@@ -418,8 +418,9 @@ void omap_uart_resume_idle(int num)
 			}
 
 			/* Check for normal UART wakeup */
-			if (__raw_readl(uart->wk_st) & uart->wk_mask)
-				omap_uart_block_sleep(uart);
+			if (uart->wk_st && uart->wk_mask)
+				if (__raw_readl(uart->wk_st) & uart->wk_mask)
+					omap_uart_block_sleep(uart);
 			return;
 		}
 	}
-- 
1.6.6.1

